src/client/Client.js

```diff
@@ -492,6 +492,12 @@ class Client extends BaseClient {
     if (typeof options.retryLimit !== 'number' || isNaN(options.retryLimit)) {
       throw new TypeError('CLIENT_INVALID_OPTION', 'retryLimit', 'a number');
     }
+    if (
+      typeof options.disableRateLimitQueue !== 'undefined' &&
```

Comment:
I think this name is misleading, maybe something like `rejectOnRateLimit` or similar might be better?

Reply:
yeah im not too happy with the current name either but i didnt come up with anything better. I think rejectOnRateLimit is nice, or maybe shouldRejectOnRateLimit?

Reply:
`rejectOnRateLimit` is fine.